components/function-controller/internal/controllers/serverless/configmap.go

```diff
@@ -4,98 +4,106 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/go-logr/logr"
-	"github.com/kyma-project/kyma/components/function-controller/internal/controllers/serverless/runtime"
-	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	apilabels "k8s.io/apimachinery/pkg/labels"
 
 	serverlessv1alpha1 "github.com/kyma-project/kyma/components/function-controller/pkg/apis/serverless/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func (r *FunctionReconciler) createConfigMap(ctx context.Context, log logr.Logger, instance *serverlessv1alpha1.Function, rtm runtime.Runtime) error {
-	configMap := r.buildConfigMap(instance, rtm)
+func stateFnInlineCheckSources(ctx context.Context, r *reconciler, s *systemState) stateFn {
+
+	labels := s.internalFunctionLabels()
 
-	log.Info("CreateWithReference ConfigMap")
-	if err := r.client.CreateWithReference(ctx, instance, &configMap); err != nil {
-		log.Error(err, "Cannot create ConfigMap")
-		return err
+	r.err = r.client.ListByLabel(ctx, s.instance.GetNamespace(), labels, &s.configMaps)
+	if r.err != nil {
+		return nil
 	}
-	log.Info(fmt.Sprintf("ConfigMap %s created", configMap.GetName()))
 
-	return r.updateStatusWithoutRepository(ctx, instance, serverlessv1alpha1.Condition{
-		Type:               serverlessv1alpha1.ConditionConfigurationReady,
-		Status:             corev1.ConditionTrue,
-		LastTransitionTime: metav1.Now(),
-		Reason:             serverlessv1alpha1.ConditionReasonConfigMapCreated,
-		Message:            fmt.Sprintf("ConfigMap %s created", configMap.GetName()),
-	})
+	r.err = r.client.ListByLabel(ctx, s.instance.GetNamespace(), labels, &s.deployments)
+	if r.err != nil {
+		return nil
+	}
+
+	srcChanged := s.inlineFnSrcChanged(r.cfg.docker.PullAddress)
```

Comment:
This is not used anywhere else, can we just use the function call or declare this in the `if` scope?